materialize-postgres/.snapshots/TestSQLGeneration

```diff
@@ -3,13 +3,13 @@
 CREATE TABLE IF NOT EXISTS "a-schema".target_table (
 		key1 BIGINT NOT NULL,
 		"key!2" BOOLEAN NOT NULL,
-		"nullableKey" TEXT,
+		"nullableKey" TEXT NOT NULL,
```

Comment:
We used to create key columns as nullable if they had `null` in their type, was this expected?

Reply:
Key columns were previously allowed to be null if the input schema permitted nulls, regardless of whether they were singular or compound. I'm pretty sure (though not 100% certain) that was deliberate.

Reply:
Hmm, but flow itself does not allow keys to be nullable, right? I can't think of a scenario where a materialization connector will actually be able to materialize a document with nullable keys 🤔

Reply:
I believe it is planned for keys to be nullable in flow. The new `materialize-postgres` connector has [some code] (https://github.com/estuary/connectors/blob/e2526149ab5ad42a47bb8a7eefc3e113b722012e/materialize-postgres/sqlgen.go#L187-L191) to allow that even, although it is not supported in the runtime yet. Also see https://github.com/estuary/flow/discussions/643 for other background discussion about nullable keys.

There may be an argument for not applying the `not null` requirement to any column at all at this point, even primary keys. If we implemented nullable key support prior to materialization schema migration, we could end up with broken materializations because a schema key gets changed to nullable. But if we get around to implementing schema migration before nullable keys it wouldn't matter. One caveat is that I'm pretty sure in the case of postgres at least setting the primary key automatically sets `not null` if the primary key is a single column; not sure how it works for composite keys.